src/js-display-panel.ts

```diff
@@ -0,0 +1,135 @@
+import { QueryStringReader } from './github/gtmsportswear/js-utilities@1.0.0/js-utilities';
+
+export abstract class DisplayPanel {
+  public draw(): Element {
+    const panel = document.createElement('div');
+    panel.classList.add('display-panel');
+
+    return panel;
+  }
+}
+
+export class FeatureToggleDisplayPanel extends DisplayPanel {
+  constructor(private toggles: ToggleStatus[]) {
+    super();
+  }
+
+  public draw(): Element {
+    const panel = super.draw(),
+          list = this.createToggleList();
+    
+    panel.appendChild(this.createTitleNode());
+    panel.appendChild(list);
+
+    this.toggles.forEach(toggle => {
+      list.appendChild(this.createToggleListItem(toggle));
+    });
+
+    panel.appendChild(this.createHidePanelButton());
+
+    return panel;
+  }
+
+  private createTitleNode(): Element {
+    const titleNode = document.createElement('h2');
+    titleNode.innerHTML = 'Feature Toggles';
+
+    return titleNode;
+  }
+
+  private createHidePanelButton(): Element {
+    const hidePanelButton = document.createElement('button');
+    hidePanelButton.classList.add('display-panel__hide');
+    hidePanelButton.innerHTML = 'Hide Panel';
+
+    return hidePanelButton;
+  }
+
+  private createToggleList(): HTMLUListElement {
+    return document.createElement('ul');
+  }
+
+  private createToggleListItem(toggle: ToggleStatus): HTMLLIElement {
+    const li = document.createElement('li');
+    li.appendChild(this.createToggleStatusNode(toggle));
+    li.appendChild(this.createToggleNameNode(toggle.Name));
+
+    return li;
+  }
+
+  private createToggleNameNode(name: string): Element {
+    const nameNode = document.createElement('div');
+    nameNode.classList.add('item-name');
+    nameNode.innerHTML = name;
+
+    return nameNode;
+  }
+
+  private createToggleStatusNode(toggle: ToggleStatus): Element {
+    const featureToggleUpdateLink = this.createFeatureToggleLink(toggle);
+    
+    const statusNode = document.createElement('label');
+    statusNode.classList.add('item-status');
+    const toggleCheckbox = document.createElement('input');
+    toggleCheckbox.setAttribute('type', 'checkbox');
+    
+    const sliderDisplayNode = document.createElement('span');
+    sliderDisplayNode.classList.add('slider');
+    sliderDisplayNode.setAttribute('data-reload-href', featureToggleUpdateLink);
+    sliderDisplayNode.addEventListener('click', e => {
+      e.preventDefault();
+
+      this.reloadPageWithUpdatedQueryString(featureToggleUpdateLink);
+    });
+
+    statusNode.appendChild(toggleCheckbox);
+    statusNode.appendChild(sliderDisplayNode);
+
+    if (toggle.IsActive)
+      toggleCheckbox.setAttribute('checked', 'true');
+    else
+      toggleCheckbox.removeAttribute('checked');
+
+    return statusNode;
+  }
+
+  private createFeatureToggleLink(toggle: ToggleStatus): string {
+    const featureOnOffString = toggle.IsActive ? 'featureoff' : 'featureon',
```

Comment:
@StentorianCat great observation. We are creating the link which will toggle the feature to the opposite of the current state. Which is why if the toggle is active, we want the link to point to `featureoff` so that clicking it turns off the feature.